torch/csrc/distributed/autograd/context/context.cpp

```diff
@@ -211,6 +213,15 @@ const c10::Dict<torch::Tensor, torch::Tensor> DistAutogradContext::
   return accumulatedGrads_;
 }
 
+void DistAutogradContext::runGradCallbaclForVariable(
+    const torch::autograd::Variable& variable,
+    GradCallback&& cb) {
+  std::lock_guard<std::mutex> guard(lock_);
+  auto it = accumulatedGrads_.find(variable);
+  TORCH_INTERNAL_ASSERT(it != accumulatedGrads_.end());
```

Comment:
Let's add an error message here

Reply:
question: I did not follow previous diffs, just curious whether accumulatedGrads_ always contains the variable when a grad is ready? based on my previous understanding, accumulatedGrads_ will contains grads only when all grads are ready?

Reply:
> question: I did not follow previous diffs, just curious whether accumulatedGrads_ always contains the variable when a grad is ready? based on my previous understanding, accumulatedGrads_ will contains grads only when all grads are ready?

This function will be called only after the post hook in the accumulateGrad node is called. That guarantees the grad is ready.  `accumulatedGrads_` now accumulates the grad of a variable once it's ready after RP #33951